services/wallet/router/router_v2.go

```diff
@@ -221,8 +236,13 @@ func findBestV2(routes [][]*PathV2, tokenPrice float64, nativeChainTokenPrice fl
 		for _, path := range route {
 			tokenDenominator := big.NewFloat(math.Pow(10, float64(path.FromToken.Decimals)))
 
-			path.requiredTokenBalance = new(big.Int).Set(path.AmountIn.ToInt())
+			path.requiredTokenBalance = big.NewInt(0)
 			path.requiredNativeBalance = big.NewInt(0)
+			if path.FromToken.IsNative() {
+				path.requiredNativeBalance.Add(path.requiredNativeBalance, path.AmountIn.ToInt())
+			} else {
+				path.requiredTokenBalance.Add(path.requiredTokenBalance, path.AmountIn.ToInt())
+			}
```

Comment:
Who those change?

Also I see this is duplicated , maybe we could do:

path.UpdateRequiredBalance()

Reply:
> why those changes?

Because there is a bonder fee for the native token as well, not only for erc20 tokens.

> Also I see this is duplicated , maybe we could do...

It's not duplicated, it adds amount to different variables.